### PR TITLE
Refactor generator

### PR DESCRIPTION
**Pull request by MaxNoe**
I will keep you updated here on the progress to implement a only-forward seeking eventio object generator here.

### CODE REVIEW

File: eventio/file_types.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    '''
+    with open(path, 'rb') as f:
+        marker_bytes = f.read(2)
+
+    return marker_bytes[0] == 0x1f and marker_bytes[1] == 0x8b
+
+
+def is_eventio(path):
+    '''
+    Test if a file is an eventio file by checking it's first two bytes
```
Comment (dneise) [inline eventio/file_types.py]: lying docstring :-D ... `first four bytes` not two

Comment (MaxNoe) [reply eventio/file_types.py]: yay, copy paste

Comment (dneise) [reply eventio/file_types.py]: Document intention not implementation :-P

File: eventio/tests/test_bits.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    assert bool_bit_from_pos(word, 5)
+    assert not bool_bit_from_pos(word, 6)
+
+
+def test_get_bits_from_word():
+    from eventio.bits import get_bits_from_word
+
+    word = 0b0000_0000_0000_0000_1010_1100_0000_0000
+
+    assert get_bits_from_word(word, num_bits=4, first=10) == 0b1011
```
Comment (dneise) [inline eventio/tests/test_bits.py]: I like these new tests here. They document clearly.

File: eventio/tests/test_constants.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        SYNC_MARKER_LITTLE_ENDIAN,
+        SYNC_MARKER_BIG_ENDIAN,
+        SYNC_MARKER_UINT8_VALUE,
+        SYNC_MARKER_INT8_VALUE
+    )
+
+    assert struct.unpack('<L', SYNC_MARKER_LITTLE_ENDIAN)[0] == SYNC_MARKER_UINT8_VALUE
+    assert struct.unpack('<l', SYNC_MARKER_LITTLE_ENDIAN)[0] == SYNC_MARKER_INT8_VALUE
+    assert struct.unpack('>L', SYNC_MARKER_BIG_ENDIAN)[0] == SYNC_MARKER_UINT8_VALUE
+    assert struct.unpack('>l', SYNC_MARKER_BIG_ENDIAN)[0] == SYNC_MARKER_INT8_VALUE
```
Comment (dneise) [inline eventio/tests/test_constants.py]: +1

File: eventio/tests/test_file_recognition.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            f.write(b'lkdlsandlnl3nlasndla')
+
+        assert not is_gzip(tmp.name)
+
+
+def test_is_eventio():
+    from eventio.file_types import is_eventio
+
+    assert is_eventio(testfile)
+    assert is_eventio(testfile_gz)
```
Comment (dneise) [inline eventio/tests/test_file_recognition.py]: all good here

File: eventio/tests/test_objects.py [hunk 5]
```diff
@@ -8,29 +8,20 @@
...
+        obj = next(f)
 
-def test_level():
-    f = eventio.EventIOFile(testfile)
+        obj.seek(0)
+        pos = obj.seek(0, 2)
 
-    assert f[0].header.level == 0
-    assert f[5][0].header.level == 1
+        assert pos == obj.header.length
```
Comment (dneise) [inline eventio/tests/test_objects.py]: got it

**Pull request merged**